<launch>
  <arg name="manager" value="visp_nodelet_manager" />

  <!-- FIXME: do not use the timestamps from the bag file,
    it makes the tracker hang for some reason. -->
  <param name="use_sim_time" value="true" />

   <!-- nodelet manager process -->
  <node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" />

  <!-- Launch the tracking node -->
  <node pkg="nodelet" type="nodelet" name="tracker"
        args="load visp_tracker/Tracker $(arg manager)">
    <param name="camera_prefix" value="/depth_camera/depth" />
  </node>

  <!-- Launch the client (GUI).

       It is mandatory to provide an initial estimation of the object
       position in order to start the tracking. To achieve this, this
       graphical tool can be used. It is not required to call it
       through this launch file, you can also directly call the
       initialization service to provide the initial estimation.

       The model_path variable can either be a standard Unix path or a
       path using the resource_retriever syntax.
       See http://ros.org/wiki/resource_retriever

    -->
  <node pkg="nodelet" type="nodelet" name="client"
        args="load visp_tracker/TrackerClient $(arg manager)">
    <param name="model_path" value="package://visp_tracker/models" />
    <param name="model_name" value="laas-box" />

    <!-- Load recommended settings for tracking initialization. They
      will be automatically forwarded to the tracking node if the
      initialization succeed. -->
    <rosparam file="$(find visp_tracker)/models/laas-box/tracker.yaml" />
  </node>

  <!-- Launch the viewer (GUI)

       This is totally optional. When the tracker is running on a
       remote computer, it is highly recommended to rely on
       image_transport to stream compressed images. You can use the
       ~image_transport parameter to do so, see the image_transport
       package documentation for more information.
  -->
  <node pkg="nodelet" type="nodelet" name="viewer"
        args="load visp_tracker/TrackerViewer $(arg manager)" />
</launch>